src/util/util_env.cpp

```diff
@@ -46,27 +59,38 @@ namespace dxvk::env {
 
   std::string getExeBaseName() {
     auto exeName = getExeName();
+#ifdef _WIN32
     auto extp = matchFileExtension(exeName, "exe");
 
     if (extp != std::string::npos)
       exeName.erase(extp);
+#endif
 
     return exeName;
   }
 
 
   std::string getExePath() {
+#if defined(_WIN32)
     std::vector<WCHAR> exePath;
     exePath.resize(MAX_PATH + 1);
 
     DWORD len = ::GetModuleFileNameW(NULL, exePath.data(), MAX_PATH);
     exePath.resize(len);
 
     return str::fromws(exePath.data());
+#elif defined(__linux__)
```

Comment:
`#else`

Also, there is so many platform-specific blocks in this file, maybe better to use whole separate blocks for linux and _WIN32, or surround whole function in block, or use `if constexpr ...` to make code less ugly.

Reply:
This is not an "else", this is linux specific code. The others would work on BSD, this would not.